arangod/RocksDBEngine/RocksDBTransactionState.cpp

```diff
@@ -318,20 +318,28 @@ arangodb::Result RocksDBTransactionState::internalCommit() {
     }
 #endif
 
-    ++_numCommits;
+    // total number of sequence ID consuming records
+    uint64_t numOps = _rocksTransaction->GetNumPuts() +
+                      _rocksTransaction->GetNumDeletes() +
+                      _rocksTransaction->GetNumMerges();
+    // will invaliate all counts
     result = rocksutils::convertStatus(_rocksTransaction->Commit());
-
+    
     if (result.ok()) {
-      rocksdb::SequenceNumber latestSeq =
-        rocksutils::globalRocksDB()->GetLatestSequenceNumber();
+      TRI_ASSERT(numOps > 0); // simon: should hold unless we're beeing stupid
+      rocksdb::SequenceNumber postCommitSeq = _rocksTransaction->GetCommitedSeqNumber();
+      if (ADB_LIKELY(numOps > 0)) {
+        postCommitSeq += numOps - 1; // add to get to the next batch
+      }
```

Comment:
I must admit I'm a bit lost. Why are we predicting a future sequence number here?

Reply:
because we can!

Reply:
No, I mean, why do we want to use the future value to commit the counts? The future value isn't actually committed yet, and it's possible we'll have other transactions which could affect those counts in between, right?

Reply:
this is the sequence number immediately _after_ the last operation in the WAL. The latest tick is always greater than postCommitSeq (which I assert on in the next line)

Reply:
Oh, I was under the impression that `seq_used` was the highest sequence number written by the transaction/batch. That makes more sense then. Okay.